modules/bullet/space_bullet.cpp

```diff
@@ -582,36 +585,36 @@ void SpaceBullet::create_empty_world(bool p_create_soft_world) {
 	if (p_create_soft_world) {
 		world_mem = malloc(sizeof(btSoftRigidDynamicsWorld));
 	} else {
-		world_mem = malloc(sizeof(btDiscreteDynamicsWorld));
+		world_mem = malloc(sizeof(btDiscreteDynamicsWorldMt));
 	}
 
 	ERR_FAIL_COND_MSG(!world_mem, "Out of memory.");
 
 	if (p_create_soft_world) {
 		collisionConfiguration = bulletnew(GodotSoftCollisionConfiguration(static_cast<btDiscreteDynamicsWorld *>(world_mem)));
 	} else {
-		collisionConfiguration = bulletnew(GodotCollisionConfiguration(static_cast<btDiscreteDynamicsWorld *>(world_mem)));
+		collisionConfiguration = bulletnew(GodotCollisionConfiguration(static_cast<btDiscreteDynamicsWorldMt *>(world_mem)));
 	}
 
 	dispatcher = bulletnew(GodotCollisionDispatcher(collisionConfiguration));
 	broadphase = bulletnew(btDbvtBroadphase);
 	solver = bulletnew(btSequentialImpulseConstraintSolver);
+	int32_t process_count = OS::get_singleton()->get_processor_count() - 2;
```

Comment:
Could you explain your reasoning for hardcoding the number of used threads? I am not sure how exactly multithreading is implemented in bullet and how well it scales, but I am guessing that after some point (e.g. >16 cores) It would make more sense to keep some of them free for stuff like graphics or audio.

Reply:
I'm assuming that 2 is the number of hyperthreads that consist of one core. I then subtract it from the list of logical threads the cpu see. The main thing is to not step on the gameplay thread (Main).

Reply:
I am guessing this would work fine as long as there are not too many other threads out there. However, this might fail terribly if a dev writes a heavily parallelized game. In the end, it all comes down to how the threads are distributed.
I personally would probably set a max limit of physics threads, but tbh it probably won't ever really matter.

Reply:
If we end up integrating this feature, the best would be to have the max number of threads in project settings. As you mentioned, the right value really depends on how threads are used for the whole project.

Reply:
The project configuration will default to 0 and any integer number above 0 will be the limit of core threads.

Reply:
Yeah, some sort of config option would be nice. However, I am not sure if we want to use integers since the dev has no way of knowing what hardware players have. It's not a perfect solution, but what about using percentages?

Reply:
Sure a percent where it is a float number from 0 to 1? Have to see how other settings handle this.